metagraph/src/graph/annotated_dbg.cpp

```diff
@@ -376,6 +377,114 @@ AnnotatedDBG::get_label_count_quantiles(std::string_view sequence,
     return label_quantiles;
 }
 
+std::vector<std::pair<std::string, std::vector<SmallVector<uint64_t>>>>
+AnnotatedDBG::get_kmer_coordinates(std::string_view sequence,
+                                   size_t num_top_labels,
+                                   double presence_ratio) const {
+    assert(presence_ratio >= 0.);
+    assert(presence_ratio <= 1.);
+    assert(check_compatibility());
+
+    if (sequence.size() < dbg_.get_k())
+        return {};
+
+    std::vector<node_index> path;
+    size_t num_kmers = sequence.size() - dbg_.get_k() + 1;
+    path.reserve(num_kmers);
+
+    graph_->map_to_nodes(sequence, [&](node_index i) {
+        path.push_back(i);
+    });
+
+    return get_kmer_coordinates(path, num_top_labels, presence_ratio);
+}
+
+std::vector<std::pair<std::string, std::vector<SmallVector<uint64_t>>>>
+AnnotatedDBG::get_kmer_coordinates(const std::vector<node_index> &path,
+                                   size_t num_top_labels,
+                                   double presence_ratio) const {
+    assert(presence_ratio >= 0.);
+    assert(presence_ratio <= 1.);
+    assert(check_compatibility());
+
+    if (!path.size())
+        return {};
+
+    std::vector<row_index> rows;
+    rows.reserve(path.size());
+
+    std::vector<size_t> ids;
+    ids.reserve(path.size());
+
+    for (node_index i : path) {
+        if (i > 0) {
+            ids.push_back(rows.size());
+            rows.push_back(graph_to_anno_index(i));
+        } else {
+            ids.push_back(-1);
+        }
+    }
+
+    uint64_t min_count = std::max(1.0, std::ceil(presence_ratio * path.size()));
+    if (rows.size() < min_count)
+        return {};
+
+    auto rows_tuples = dynamic_cast<const MultiIntMatrix &>(annotator_->get_matrix())
```

Comment:
When an annotator not supporting coordinates is loaded, this throws a `std::bad_cast` without giving more useful feedback. Either catch this is `load_annotated_graph`, or here and rethrow it with a more useful message.

Reply:
done